server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -463,6 +467,115 @@ describe("Routerlicious", () => {
 						]);
 						await Promise.all(promises);
 					});
+
+					it("Verify v2 signals", async () => {
+						let userSignalCounts = [0, 0, 0];
+
+						const createSignalPromise = (
+							userIndex: number,
+							socket: LocalWebSocket,
+						): Promise<ISignalMessage> => {
+							return new Promise((resolve) => {
+								socket.on("signal", (signal: ISignalMessage) => {
+									userSignalCounts[userIndex]++;
+									resolve(signal);
+								});
+							});
+						};
+						const firstSocket = webSocketServer.createConnection();
+
+						const firstConnectMessage = await connectToServer(
+							testId,
+							testTenantId,
+							testSecret,
+							firstSocket,
+							true,
+						);
+
+						const secondSocket = webSocketServer.createConnection();
+
+						const secondConnectMessage = await connectToServer(
+							testId,
+							testTenantId,
+							testSecret,
+							secondSocket,
+							true,
+						);
+
+						const thirdSocket = webSocketServer.createConnection();
+
+						await connectToServer(testId, testTenantId, testSecret, thirdSocket, true);
+
+						const user1SignalPromise = createSignalPromise(0, firstSocket);
+						const user2SignalPromise = createSignalPromise(1, secondSocket);
+						const user3SignalPromise = createSignalPromise(2, thirdSocket);
+
+						const targetedSignal: ISentSignalMessage = {
+							targetClientId: firstConnectMessage.clientId,
+							content: "TestSignal",
+						};
+
+						secondSocket.send("submitSignal", secondConnectMessage.clientId, [
```

Comment:
Could this be a separate test case? Or you want to make sure this particular sequence of "using targeted signals", then "using broadcast" works as a single test case?

Reply:
I grouped v2 signal testing together but I can definitely split these up